### DISCUSSION TITLE
Microsoft.Tools.WindowsInstallerXml.AssemblyDefaultWixExtensionAttribute' was not defined

### DISCUSSION THREAD

**Discussion by fng2** [wixtoolset/issues#7557] [Questions]
This has probably come up before but despite my googling and searching here I could not find an answer.
I am building an excel VSTO addin using VS2015 and Wix 3.11. Same setup I've used several times. After adding the default Wix setup project and building the solution I get the above error. In the Wix project I have added references to the the addin dll (which builds fine), the wixUIExtension, the wixUtilExtension and product.wxs. These are the same steps I have used before without issue so I am not sure why wix is not able to use any of it's built in variables. It is probably something simple, but eluding me now.
here is my product.wxs:
`<?xml version="1.0" encoding="UTF-8"?>
<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
	<Product Id="312BB152-626D-4380-AD94-03F387BE38E8" Name="HIRFSetup" Language="1033" Version="1.0.0.0" Manufacturer="Garmin" UpgradeCode="c641e239-0c94-4918-9c3f-61ab792f0dbf">
		<Package InstallerVersion="200" Compressed="yes" InstallScope="perMachine" />

		<MajorUpgrade DowngradeErrorMessage="A newer version of [ProductName] is already installed." />
    <MediaTemplate EmbedCab="yes" />
    <UIRef Id="WixUI_Mondo" />
    
    <WixVariable Id="WixUILicenseRtf" Value="License.rtf" />

		<Feature Id="ProductFeature" Title="HIRFSetup" Level="1">
			<ComponentGroupRef Id="ProductComponents" />
		</Feature>
	</Product>

	<Fragment>
		<Directory Id="TARGETDIR" Name="SourceDir">
			<Directory Id="ProgramFilesFolder">
				<Directory Id="INSTALLFOLDER" Name="HIRFSetup" />
			</Directory>
		</Directory>
	</Fragment>

	<Fragment>
		<ComponentGroup Id="ProductComponents" Directory="INSTALLFOLDER">
			
			
				
			

      
      <Component>
        <File Source="$(var.HIRFAddin.TargetDir)Microsoft.Office.Tools.Common.v4.0.Utilities.dll"></File>
      </Component>

    </ComponentGroup>
	</Fragment>
</Wix>`

Thanks for any pointers !

**Comment by barnson** [comment]
> added references to the the addin dll 

Don't do that. References are WiX extension DLLs.

**Comment by fng2** [comment]
I misspoke, I  did not add a reference to the addin.dll, it was a reference to the addin project. This is how it is on all my other projects which work fine and removing it does not help. Wix is still unaware of it's built in variables.

**Comment by barnson** [comment]
The error message says you have a reference to a DLL that isn't a WiX extension. Check the .wixproj.

**Comment by fng2** [comment]
Ok, I found the problem. My addin was named "HIRFAddin" or so I thought. When I created the project in VS I miss typed "HIRFAddIn". Note the capitol "I" in addin. My references were defined as using "HIRFAddin" and Wix being very case sensitive was not registering them. Once I fixed that all was well.
Thanks
